### ISSUE TITLE
Option to make Change layout the default

### ISSUE THREAD

**Issue by dromer** [Houston4444/HoustonPatchbay#1]
Currently you have to set "Change layout" individually for each program, plugin, device etc.

Since I want to have this for pretty much everything I do in the patchbay it would be great if this can be set as default option in the settings.

**Comment by Houston4444** [comment]
Interesting.
"Change layout" is not a state, it is an action.
The idea of this feature is of course to optimize box sizes.
In memory, there are 3 possible values for a box layout: Auto, High, Large.
The default value is 'auto', in this mode the selected layout is this one which gives the smallest box area (high or large).
for High or Large values, the layout is simply forced to high or large.

If you click on 'Change layout', suppose the box layout mode was 'auto' but displayed in 'high' mode, then the box layout will change from 'auto' to 'large'.

Note that the same mechanism is used for boxes with inputs and outputs and boxes with only inputs or outputs. For boxes with only inputs or outputs, 'High' means Title on top, 'Large' means title on side.

What I could do in options is two sliders, one for boxes with inputs and outputs, and one for others. It could determinate what the 'auto' mode does, from "always large" to "always high", applying a factor to one of the areas to encourage one mode on the other.

Maybe theses options are overrated... Probably pure area comparison is not a good default mode.

**Comment by dromer** [comment]
Ah I see. So far it seemed that auto would always default to the layout I didn't like (is that "high"? I don't know. with ports interlaced and the smallest "width").

**Comment by Houston4444** [comment]
![hydro_large](https://user-images.githubusercontent.com/3808689/187064067-51325b7b-74c6-48db-97f9-59b0024340b9.png)
![hydro_high](https://user-images.githubusercontent.com/3808689/187064063-216426fa-57e2-4031-b4c5-6401c3c2d34d.png)
This is the main reason of a such feature.

**Comment by dromer** [comment]
Yeah I understand, but it's also inconsistent. See the ordering of midi i/o for these devices:
![](https://i.imgur.com/pDI965P.png)

On the left it starts with the input, in the right with the output.

But for both I prefer to have them on the same level, which makes it much easier to see what the port layout is (for me).

**Comment by Houston4444** [comment]
Ok, The port order is given by JACK. Here the order inconsistency comes from a2j or pipewire. Patchage does the same:

![Screenshot_20220828_104448](https://user-images.githubusercontent.com/3808689/187065640-b8c3165f-0171-4fea-9bf0-e7423af5af31.png)

I'll put the options, I think you are right and the default should be to prefer 'large' layout except if the area win is important (for boxes with inputs and outputs).

Any problem with boxes with title on side (splitted boxes) ?

**Comment by Houston4444** [comment]
Ok, you will find a combobox option in tricks tab !
